src/experimental/tools/download2go/index.ts

```diff
@@ -0,0 +1,415 @@
+      delete activeDownloads[contentID];
+      delete activePauses[contentID];
+    } catch (e) {
+      const error = e as Error | undefined;
+      this.trigger("error", {
+        action: "pause",
+        contentID,
+        error: error !== undefined ? error : new Error("A Unexpected error happened"),
+      });
+    }
+  }
+
+  /**
+   * Get all the downloaded entry (manifest) partially or fully downloaded.
+   * @returns {Promise.<IStoredManifest[]|void>}
+   */
+  getAllOfflineContent(
+    limit?: number
+  ): Promise<IStoredManifest[] | undefined> | void {
+    try {
+      if (this.db == null) {
+        throw new Error("The IndexDB database has not been created!");
+      }
+      // TODO: return formatted content ?
+      return this.db.getAll("manifests", undefined, limit);
+    } catch (e) {
+      const error = e as Error;
+      this.trigger("error", {
+        action: "getAllDownloadedMovies",
+        error: new IndexDBError(error.message),
+      });
+    }
+  }
+
+  /**
+   * Get a singleMovie ready to be played by the rx-player,
+   * could be fully or partially downloaded.
+   * @param {string} contentID
+   * @returns {Promise.<IContentLoader|void>}
+   */
+  async getSingleContent(contentID: string): Promise<IContentLoader | void> {
+    try {
+      if (this.db == null) {
+        throw new Error("The IndexDB database has not been created!");
+      }
+      const [contentManifest, contentsProtection]: [
+        IStoredManifest?,
+        IContentProtection[]?,
+      ] = await PPromise.all([
+        this.db.get("manifests", contentID),
+        this.db
+          .transaction("contentsProtection", "readonly")
+          .objectStore("contentsProtection")
+          .index("contentID")
+          .getAll(IDBKeyRange.only(contentID)),
+      ]);
+      if (contentManifest === undefined || contentManifest.manifest === null) {
+        throw new SegmentConstuctionError(
+          `No Manifest found for current content ${contentID}`
+        );
+      }
+      const {
+        progress,
+        transport,
+        size,
+        metaData,
+        manifest,
+        duration,
+      } = contentManifest;
+      const contentProtection = getKeySystemsSessionsOffline(contentsProtection);
+
+      return {
+        progress,
+        size,
+        transport,
+        contentID,
+        metaData,
+        contentProtection,
+        offlineManifest: offlineManifestLoader(
+          manifest,
+          getBuilderFormattedForAdaptations(contentManifest),
+          getBuilderFormattedForSegments(contentManifest),
+          { contentID, duration, isFinished: progress.percentage === 100, db: this.db }
+        ),
+      };
+    } catch (e) {
+      const error = e as Error | undefined;
+      this.trigger("error", {
+        action: "getSingleMovie",
+        contentID,
+        error: error !== undefined ? error : new Error("A Unexpected error happened"),
+      });
+    }
+  }
+
+  async getAvailableSpaceOnDevice() {
```

Comment:
We never use the context here, are we sure we don't want to set it as a static method?

Also, why the `OnDevice`?

Reply:
yea it makes more sense to make it `static`